engine/src/test/java/com/redhat/cloud/notifications/routers/DailyDigestResourceTest.java

```diff
@@ -0,0 +1,143 @@
+        Mockito.when(this.environment.isEnvironmentLocal()).thenReturn(false);
+        Mockito.when(this.environment.isEnvironmentStage()).thenReturn(false);
+
+        final TriggerDailyDigestRequestDto triggerDailyDigestRequestDto = new TriggerDailyDigestRequestDto(
+            "application-name",
+            "bundle-name",
+            "organization-id",
+            null,
+            null
+        );
+
+        // Call the endpoint under test.
+        final String response = given()
+            .basePath(API_INTERNAL)
+            .when()
+            .contentType(JSON)
+            .body(Json.encode(triggerDailyDigestRequestDto))
+            .post("/daily-digest")
+            .then()
+            .statusCode(400)
+            .extract()
+            .asString();
+
+        Assertions.assertEquals("the daily digests can only be triggered in the stage environment", response, "unexpected error message received");
+    }
+
+    /**
+     * Tests that when calling the "trigger daily digest" endpoint, a proper
+     * Kafka message, with an aggregation command, is sent.
+     * @throws IOException if the incoming message from Kafka cannot be
+     * deserialized.
+     */
+    @Test
+    void testTriggerDailyDigest() throws IOException {
+        final String applicationName = "application-name";
+        final String bundleName = "bundle-name";
+        final String orgId = "org-id";
+        final LocalDateTime end = LocalDateTime.now();
+        final LocalDateTime start = end.minusDays(5);
+
+        final TriggerDailyDigestRequestDto triggerDailyDigestRequestDto = new TriggerDailyDigestRequestDto(
+            applicationName,
+            bundleName,
+            orgId,
+            start,
+            end
+        );
+
+        // Simulate that we are in the stage environment.
+        Mockito.when(this.environment.isEnvironmentStage()).thenReturn(true);
+
+        // Call the endpoint under test.
+        given()
+            .basePath(API_INTERNAL)
+            .when()
+            .contentType(ContentType.JSON)
+            .body(Json.encode(triggerDailyDigestRequestDto))
+            .post("/daily-digest")
+            .then()
+            .statusCode(204);
+
+        // We should receive the aggregation triggered by the REST call.
+        InMemorySink<String> aggregationsOut = this.inMemoryConnector.sink(DailyDigestResource.AGGREGATION_OUT_CHANNEL);
+
+        // Make sure that the message was received before continuing.
+        Awaitility.await().until(
+            () -> aggregationsOut.received().size() == 1
+        );
+
+        final var aggregationCommands = aggregationsOut.received();
+
+        // Only one test aggregation command should have been sent to Kafka.
+        final var expectedAggregationsCommandCount = 1;
+        Assertions.assertEquals(expectedAggregationsCommandCount, aggregationCommands.size(), "unexpected number of aggregation commands sent to Kafka");
```

Comment:
That's already asserted by line 118.

Reply:
All right! I removed the redundant assertion!